datafusion/src/physical_plan/union.rs

```diff
@@ -37,23 +37,31 @@ use crate::{
 };
 use async_trait::async_trait;
 
-/// UNION ALL execution plan
+/// UNION execution plan
 #[derive(Debug)]
 pub struct UnionExec {
     /// Input execution plan
     inputs: Vec<Arc<dyn ExecutionPlan>>,
     /// Execution metrics
     metrics: ExecutionPlanMetricsSet,
+    /// Union ALL or Union
+    is_all: bool,
```

Comment:
Union all and union distinct are quite different, so if we need to add / change implementations, I think it makes sense to add a new node instead, like `UnionDistinctExec`